### PR TITLE
♻️ move some code to Board

### PR DESCRIPTION
**Pull request by garak**
This moves some code to correct responsibility.

### CODE REVIEW

File: src/Board.php [hunk 1]
```diff
@@ -97,4 +102,49 @@ public function jsonSerialize()
     {
         return json_encode($this->squares);
     }
+
+    public function ascii(): string
```
Comment (akondas) [inline src/Board.php:106]: It seems to me that this is not the board's responsibility. I would expect rather `AsciiPrinter` which implements:
```php
interface BoardPrinter
{
    public function print(Board $board): string
}
```

You can still work on the name itself. Examples: `Logger` or `Exporter`

Comment (garak) [reply src/Board.php:106]: I agree.
I just would to move it away from main class in a quick (and maybe dirty) way.

Comment (garak) [reply src/Board.php]: Added some classes to handle output (and squashed)

**Comment by garak** [comment]
It looks like there's a problem with buddy, maybe because I updated composer deps?

**Comment by akondas** [comment]
You can check details in build log:
```
composer validate
./composer.json is valid
The lock file is not up to date with the latest changes in composer.json, it is recommended that you run `composer update` or `composer update <package name>`.
```

Looks like composer.json and composer.lock aren't in sync

**Comment by garak** [comment]
> Looks like composer.json and composer.lock aren't in sync

That's weird, since I committed both.
Can you try to checkout this PR and see if you're able to do `composer install` in your local env?

**Comment by akondas** [comment]
Abel to install doesn't mean that lock is synced

wt., 21 kwi 2020, 13:41 użytkownik Massimiliano Arione <
notifications@github.com> napisał:

**Comment by garak** [comment]
You're right. I synced lock, now it's OK

File: src/Output/OutputInterface.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PChess\Chess\Output;
+
+use PChess\Chess\Chess;
+
+interface OutputInterface
```
Comment (akondas) [inline src/Output/OutputInterface.php:9]: :+1:

File: src/Output/UnicodeOutput.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PChess\Chess\Output;
+
+use PChess\Chess\Board;
+use PChess\Chess\Chess;
+
+final class UnicodeOutput extends BasicOutput implements OutputInterface
```
Comment (akondas) [inline src/Output/UnicodeOutput.php:10]: maybe emoji output :chess_pawn:

Comment (garak) [reply src/Output/UnicodeOutput.php:10]: unfortunately, the only piece present in emoji set is pawn.

**Pull request merged**